sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -410,7 +414,18 @@ export class ServiceBusSessionReceiverImpl implements ServiceBusSessionReceiver
       abortSignal: options?.abortSignal
     };
     return retry<ServiceBusReceivedMessage[]>(config).catch((err) => {
-      throw translateServiceBusError(err);
+      const error = translateServiceBusError(err);
+      if (
+        isServiceBusError(error) &&
+        (error.code as InternalServiceBusErrorCode) === "UnsettledMessagesLimitExceeded"
+      ) {
+        // To be consistent with other languages, not throwing for UnsettledMessagesLimitExceeded case.
+        // If the unsettled messages limit is reached, we'd just return 0 messages instead of throwing
+        // (until the service fixes the "draining" bug at the terminal case and we extend the 2048 limit in rhea)
```

Comment:
```suggestion
        // (until the service fixes the "draining" bug at the terminal case and/or we extend the 2048 limit in rhea to infinite like the .NET SDK)
```
Repeat it for sessions